shared/teams/role-picker/controlled-container.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import React from 'react'
+import {RoleOptions} from '.'
+import {connect} from 'react-redux'
+import {compose, withHandlers, withState} from 'recompose'
+import {PopupDialog, HeaderHoc, ScrollView} from '../../common-adapters/index'
+import {isMobile} from '../../constants/platform'
+import {type TypedState} from '../../constants/reducer'
+import {type TeamRoleType} from '../../constants/teams'
+
+/* 
+  Pass through via routeprops
+  onComplete gets selected role
+  selectedRole sets a default role for the component
+  allowOwner specifies whether the user can choose the 'owner' option
+*/
+export type ControlledRolePickerProps = {
+  onComplete: (role: TeamRoleType) => void,
+  selectedRole?: TeamRoleType,
+  allowOwner?: boolean,
+}
+
+const mapStateToProps = (state: TypedState, {routeProps}) => {
+  const currentType = routeProps.get('selectedRole')
+  const _onComplete = routeProps.get('onComplete')
+  const allowOwner = routeProps.get('allowOwner')
```

Comment:
I think for open teams we don't want to allow everyone to be an admin, so maybe add another flag for allowAdmin